src/realm/util/thread.cpp

```diff
@@ -95,6 +95,25 @@ void Thread::set_name(const std::string& name)
     int r = pthread_setname_np(name.data());
     if (REALM_UNLIKELY(r != 0))
         throw std::system_error(r, std::system_category(), "pthread_setname_np() failed");
+#elif defined(_WIN32)
+    // SetThreadDescription was only introduced in Windows 10 1607, so we need to dynamically load it.
+    HRESULT(WINAPI * SETTHREADDESCRIPTION)(HANDLE hThread, PCWSTR threadDescription) = nullptr;
+#if !defined(NTDDI_WIN10_RS1) || NTDDI_VERSION < NTDDI_WIN10_RS1
+    auto proc_address = GetProcAddress(GetModuleHandle(L"Kernel32.dll"), "SetThreadDescription");
+    if (!proc_address) {
+        return;
+    }
+    SETTHREADDESCRIPTION = reinterpret_cast<decltype(SETTHREADDESCRIPTION)>(proc_address);
+#else
+    SETTHREADDESCRIPTION = &SetThreadDescription;
+#endif
+    std::wstring name_wide;
+    name_wide.resize(MultiByteToWideChar(CP_UTF8, 0, name.c_str(), name.length(), nullptr, 0));
+    MultiByteToWideChar(CP_UTF8, 0, name.c_str(), name.length(), name_wide.data(), name_wide.size());
+    HRESULT result = SETTHREADDESCRIPTION(GetCurrentThread(), name_wide.data());
+    if (REALM_UNLIKELY(!SUCCEEDED(result))) {
+        throw std::system_error(result, std::system_category(), "SetThreadDescription failed");
```

Comment:
should we just return here. is setting the thread name a critical operation that we need to fail. Seems like previously we just ignore if the `proc_address` is not found

Reply:
The pthreads version above throws, so I assumed the Windows one should too. It doesn't strike me that this should be exception-worthy, but I don't know what else might depend on the current behavior so I didn't want to change it.

Reply:
ok, makes sense. resolving